ntpcheck/checker/chrony.go

```diff
@@ -17,14 +17,51 @@ limitations under the License.
 package checker
 
 import (
+	"fmt"
 	"io"
+	"net"
+	"os"
+	"path"
 
 	"github.com/facebookincubator/ntp/protocol/chrony"
 	"github.com/facebookincubator/ntp/protocol/control"
 	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 )
 
+type chronyConn struct {
+	net.Conn
+	local string
+}
+
+// Dial opens a unixgram connection with chrony
+func Dial(address string) (*chronyConn, error) {
+	base, _ := path.Split(address)
+	local := path.Join(base, fmt.Sprintf("chronyc.%d.sock", os.Getpid()))
+	conn, err := net.DialUnix("unixgram",
+		&net.UnixAddr{Name: local, Net: "unixgram"},
```

Comment:
why do we need the _laddr_? Ain't the comms over _raddr_ bi-directional?

Reply:
I tested it with a nil `laddr` and couldn't make it work.

Reply:
With same `laddr` and `raddr`:

```
dial unixgram /var/run/chrony/chronyd.sock->/var/run/chrony/chronyd.sock: bind: address already in use
```

With nil `laddr`:

```
read unixgram @->/var/run/chrony/chronyd.sock: i/o timeout
```

Reply:
Could you try with _laddr_ = _raddr_ please?